src/app/qgsfeatureaction.cpp

```diff
@@ -82,8 +82,12 @@ QgsAttributeDialog *QgsFeatureAction::newDialog( bool cloneFeature )
       if ( !action.runable() )
         continue;
 
+      if( mLayer->readOnly() && action.isEnabledOnlyWhenEditable() )
```

Comment:
Also here update to skip  the action if the layer is not isEditable(). (And the other places below too)